dashboard/client/src/common/ProfilingLink.tsx

```diff
@@ -96,3 +127,124 @@ export const CpuProfilingLink = ({
     </Link>
   );
 };
+
+export const ProfilerButton = ({
+  profilerUrl,
+  type = "",
+}: MemoryProfilingButtonProps) => {
+  const [duration, setDuration] = useState<number | null>(5);
+  const [native, setNative] = useState<boolean>(false);
+  const [leaks, setLeaks] = useState<boolean>(false);
+  const [open, setOpen] = useState<boolean>(false);
+  const [format, setFormat] = useState("flamegraph");
+
+  const handleOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  return (
+    <div>
+      <Button variant="text" color="primary" onClick={handleOpen}>
+        Profile&nbsp;Memory{type ? ` (${type})` : ""}
+      </Button>
+
+      <Dialog open={open} onClose={handleClose}>
+        <DialogTitle>Memory Profiling Config</DialogTitle>
+        <DialogContent>
+          <InputLabel id="format-label">Format</InputLabel>
+          <Select
+            labelId="format-label"
+            id="format"
+            value={format}
+            onChange={(e) => setFormat(e.target.value as string)}
+            fullWidth
+          >
+            <MenuItem value="flamegraph">Flamegraph</MenuItem>
+            <MenuItem value="table">Table</MenuItem>
+          </Select>
+          <br />
+          <InputLabel id="format-label">Duration</InputLabel>
+          <TextField
+            label="Duration"
```

Comment:
```suggestion
            label="Duration (seconds)"
```

Can we make the unit clear here?